service/storage_proxy.cc

```diff
@@ -2230,6 +2253,20 @@ future<> storage_proxy::send_to_endpoint(
             allow_hints);
 }
 
+future<> storage_proxy::send_hint_to_endpoint(frozen_mutation_and_schema fm_a_s, gms::inet_address target) {
+    if (!service::get_local_storage_service().cluster_supports_hinted_handoff_separate_connection()) {
+        return send_to_endpoint(std::move(fm_a_s), target, { }, db::write_type::SIMPLE, allow_hints::no);
+    }
+
+    return send_to_endpoint(
+            std::make_unique<hint_mutation>(std::move(fm_a_s)),
+            std::move(target),
+            {},
+            db::write_type::SIMPLE,
+            _stats,
```

Comment:
I changed the first call to use `_stats` instead, so that both use the same overload (the one with the actual logic). This way we don't need to introduce yet another overload.

Reply:
Fine with me.